app-modules/task/src/Histories/TaskHistory.php

```diff
@@ -0,0 +1,147 @@
+                'new' => TaskStatus::tryFrom($this->new[$key])?->getLabel(),
+            ],
+            'due' => [
+                'key' => 'Due',
+                'old' => data_get($this->old, $key)
+                    ? Carbon::parse($this->old[$key])->format('m-d-Y')
+                    : '(Not set)',
+                'new' => $this->new[$key]
+                    ? Carbon::parse($this->new[$key])->format('m-d-Y')
+                    : '(Not set)',
+            ],
+            'assigned_to' => [
+                'key' => 'Assigned to',
+                'old' => array_key_exists($key, $this->old)
+                    ? User::find($this->old[$key])?->name
+                    : null,
+                'new' => User::find($this->new[$key])?->name,
+                'extra' => [
+                    'old' => [
+                        'link' => array_key_exists($key, $this->old)
+                            ? UserResource::getUrl('view', ['record' => $this->old[$key]])
+                            : null,
+                    ],
+                    'new' => [
+                        'link' => UserResource::getUrl('view', ['record' => $this->new[$key]]),
+                    ],
+                ],
+            ],
+            'created_by' => [
+                'key' => 'Created by',
+                'old' => array_key_exists($key, $this->old)
+                    ? User::find($this->old[$key])?->name
+                    : null,
+                'new' => User::find($this->new[$key])?->name,
+                'extra' => [
+                    'old' => [
+                        'link' => array_key_exists($key, $this->old)
+                            ? UserResource::getUrl('view', ['record' => $this->old[$key]])
+                            : null,
+                    ],
+                    'new' => [
+                        'link' => UserResource::getUrl('view', ['record' => $this->new[$key]]),
+                    ],
+                ],
+            ],
+            default => parent::getFormattedValueForKey($key),
+        };
+    }
+
+    public function getFormattedValues(): Collection
+    {
+        return parent::getFormattedValues()->merge([
+            'concern' => [
+                'key' => 'Concern',
+                'old' => data_get($this->old, 'concern_id')
+                    ? sprintf(
+                        '%s: %s',
+                        str($this->old['concern_type'])->ucfirst(),
+                        Relation::getMorphedModel($this->old['concern_type'])::find($this->old['concern_id'])?->display_name
+                    ) : '(Not set)',
+                'new' => data_get($this->new, 'concern_id')
+                    ? sprintf(
+                        '%s: %s',
+                        str($this->new['concern_type'])->ucfirst(),
+                        Relation::getMorphedModel($this->new['concern_type'])::find($this->new['concern_id'])?->display_name
+                    ) : '(Not set)',
```

Comment:
This doesn't seem to be working as expected. When updating a task without updating the concern, I get the following:

<img width="524" alt="Screenshot 2024-04-15 at 2 58 56 PM" src="https://github.com/canyongbs/advisingapp/assets/10821263/65dbe293-9142-4c13-9078-96eaf1079ccd">

Reply:
This should be fixed now.